otwcode/otwarchive
# [WIP] AO3-5793 Docker support

# Pull Request Checklist

* [ ✔️] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [✔️] Have you read through the [contributor guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [✔️] Have you added tests for any changed functionality?
* [✔️] Have you added the [JIRA](https://otwarchive.atlassian.net) issue number as the *first* thing in your pull request title (eg: `AO3-1234 Fix thing`)
* [❌ ] Have you updated or commented on the JIRA issue with the information below? **No comment permissions**

## Issue

https://otwarchive.atlassian.net/browse/AO3-5793

## Purpose

Easier setup for local development using docker.

Side-effects:
- Had to upgrade gem mysql2 to 0.5.2 (current latest) otherwise app was crashing on startup
- Noticed that config/environments/development.rb was forcing localhost for memcache rather than reading from config

## Testing Instructions

Note that this simulates a new setup (empty DB, no dependencies available)
1.  Install Docker (https://www.docker.com/get-started)
2. Create config/database.yml with defaults:host:db
3. Create config/local.yml with 
> ES_URL: es:9200
> MEMCACHED_SERVERS: mc:11211
4. Run `docker-init.sh` (or do the commands within manually, but this seems easier)
5. Run `docker-compose up`
6. Browse to http://localhost:3000

## References

Are there any other relevant issues / PRs / mailing lists discussions related to this?

## Credit

*What name do you want us to use to credit your work in the [Archive of Our Own's Release Notes](https://archiveofourown.org/admin_posts?tag=1)?* tlee911

*What pronouns do you prefer (she/her, he/him, zie/hir etc)?* he/him
